### PR TITLE
Add fpclassify() and FP_* constants to math.pxd

### LINKED ISSUE #2514: Add fpclassify and FP_* constants to math.pxd
Would be nice to have [`fpclassify`]( https://en.cppreference.com/w/c/numeric/math/fpclassify ) and the [`FP_*` constants]( https://en.cppreference.com/w/c/numeric/math/FP_categories ) it returns defined in `math.pxd`.

### PR DESCRIPTION
**Pull request by ZackerySpytz**
Closes #2514.

### CODE REVIEW

**Comment by ZackerySpytz** [comment]
It seems like the CI failure is unrelated.

**Comment by McSinyx** [comment]
Travis also fails on master.  The funniest part is that the failing commit is actually rebased from a passing one, from GH-3391.

**Comment by da-woods** [comment]
@McSinyx I think (but not 100% sure) it's just that the particular test is slow to run and sometimes ends up hitting the 50 minute timeout.

File: Cython/Includes/libc/math.pxd [hunk 1]
```diff
@@ -104,3 +104,9 @@ cdef extern from "<math.h>" nogil:
     bint isnan(long double)
     bint isnormal(long double)
     bint signbit(long double)
+    int fpclassify(long double)
+    const int FP_NAN
```
Comment (scoder) [inline Cython/Includes/libc/math.pxd]: `const int`, or just `enum`?

Comment (ZackerySpytz) [reply Cython/Includes/libc/math.pxd]: The values of the `FP_*` constants are implementation defined.
I don't believe an enum should be used here (unless I'm missing something).

**Pull request merged**